pkg/api/validation/validation.go

```diff
@@ -2696,32 +2696,47 @@ func validateResourceQuotaScopes(resourceQuota *api.ResourceQuota) field.ErrorLi
 func ValidateResourceQuota(resourceQuota *api.ResourceQuota) field.ErrorList {
 	allErrs := ValidateObjectMeta(&resourceQuota.ObjectMeta, true, ValidateResourceQuotaName, field.NewPath("metadata"))
 
-	fldPath := field.NewPath("spec", "hard")
-	for k, v := range resourceQuota.Spec.Hard {
+	allErrs = append(allErrs, ValidateResourceQuotaSpec(resourceQuota.Spec, field.NewPath("spec"))...)
+	allErrs = append(allErrs, ValidateResourceQuotaStatus(resourceQuota.Status, field.NewPath("status"))...)
+
+	return allErrs
+}
+
+func ValidateResourceQuotaStatus(status api.ResourceQuotaStatus, fld *field.Path) field.ErrorList {
```

Comment:
The same convention was followed for the ValidatePod and ValidatePodSpec.
I guess why not pointers?

On Wed, May 25, 2016 at 4:15 PM, David Eads <notifications@github.com>
wrote:

> In pkg/api/validation/validation.go
> <https://github.com/kubernetes/kubernetes/pull/26276#discussion_r64644994>
> :
>
> > @@ -2696,32 +2696,47 @@ func validateResourceQuotaScopes(resourceQuota *api.ResourceQuota) field.ErrorLi
> >  func ValidateResourceQuota(resourceQuota *api.ResourceQuota) field.ErrorList {
> >  	allErrs := ValidateObjectMeta(&resourceQuota.ObjectMeta, true, ValidateResourceQuotaName, field.NewPath("metadata"))
> >
> > -	fldPath := field.NewPath("spec", "hard")
> > -	for k, v := range resourceQuota.Spec.Hard {
> > +	allErrs = append(allErrs, ValidateResourceQuotaSpec(resourceQuota.Spec, field.NewPath("spec"))...)
> > +	allErrs = append(allErrs, ValidateResourceQuotaStatus(resourceQuota.Status, field.NewPath("status"))...)
> > +
> > +	return allErrs
> > +}
> > +
> > +func ValidateResourceQuotaStatus(status api.ResourceQuotaStatus, fld *field.Path) field.ErrorList {
>
> take a pointer
>
> Why pointers? They aren't pointers in the API and this should never mutate.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/26276/files/77c1219806f9c21421a09386225e8d312dc7df95#r64644994>
>